claims.go

```diff
@@ -436,122 +364,180 @@ func (p *Claims) validateSwComponents() error {
 	return nil
 }
 
-func (p SwComponent) validate(idx int) error {
-	if p.MeasurementValue == nil {
-		return fmt.Errorf(
-			"invalid software-component[%d]: missing mandatory measurement-value",
-			idx,
-		)
-	}
-
-	if err := isPSAHashType(*p.MeasurementValue); err != nil {
-		return fmt.Errorf(
-			"invalid software-component[%d]: invalid measurement-value %s",
-			idx, err.Error(),
-		)
-	}
+// decorate does type enrichment on the token, to add "hidden" attributes
+// that will only be visible in the JSON (internal) encoding.
+func (c *Claims) decorate() {
+	c.decorateSecurityLifeCycle()
+	c.decoratePartitionID()
+}
 
-	if p.SignerID == nil {
-		return fmt.Errorf(
-			"invalid software-component[%d]: missing mandatory signer-id",
-			idx,
-		)
+func (c *Claims) decorateSecurityLifeCycle() {
+	// populate "_security-lifecycle-desc"
+	if c.SecurityLifeCycle != nil {
+		c.SecurityLifeCycleDesc = securityLifeCycleToString(*c.SecurityLifeCycle)
 	}
+}
 
-	if err := isPSAHashType(*p.SignerID); err != nil {
-		return fmt.Errorf(
-			"invalid software-component[%d]: invalid signer-id %s",
-			idx, err.Error(),
-		)
+func (c *Claims) decoratePartitionID() {
+	// populate "_partition-id-desc"
+	if c.PartitionID != nil {
+		c.PartitionIDDesc = partitionIDToString(*c.PartitionID)
 	}
-
-	return nil
 }
 
-func isPSAHashType(b []byte) error {
-	l := len(b)
+// ToJSON returns the (indented) JSON representation of the Claims
+func (c *Claims) ToJSON() (string, error) {
+	// add any available type enrichment
+	c.decorate()
 
-	if l != 32 && l != 48 && l != 64 {
-		return fmt.Errorf(
-			"length %d (psa-hash-type MUST be 32, 48 or 64 bytes)",
-			l,
-		)
+	buf, err := json.MarshalIndent(&c, "", "  ")
+	if err != nil {
+		return "", err
 	}
 
-	return nil
-
+	return string(buf), nil
 }
 
-// decorate does type enrichment on the token, to add "hidden" attributes
-// that will only be visible in the JSON (internal) encoding.
-func (p *Claims) decorate() {
-	p.decorateSecurityLifeCycle()
-	p.decoratePartitionID()
+// ToCBOR returns the CBOR representation of the Claims
+func (c *Claims) ToCBOR() ([]byte, error) {
+	/* No validation required here: in the emitter workflow, the PSA claims
+	 * have been validated when they got attached to the Evidence via SetClaims */
+	return em.Marshal(&c)
 }
 
-func (p *Claims) decorateSecurityLifeCycle() {
-	// populate "_security-lifecycle-desc"
-	if p.SecurityLifeCycle != nil {
-		p.SecurityLifeCycleDesc = securityLifeCycleToString(*p.SecurityLifeCycle)
+// FromCBOR takes a bytes buffer containing a PSA token and, if successful,
+// populates the receiver Claims object.
+func (c *Claims) FromCBOR(buf []byte) error {
+	err := dm.Unmarshal(buf, c)
+	if err != nil {
+		return err
 	}
-}
 
-func partitionIDToString(pid int32) string {
-	if pid <= 0 {
-		return "nspe"
-	}
-	return "spe"
+	c.decorate()
+
```

Comment:
should we not validate the claims prior to calling it a success?

Reply:
It is not possible because we don't have the validation context here.  `Claims::FromCBOR` is called by `Evidence::FromCOSE`, which has the  context to do the validation.